contrib/core-contract-tests/tests/pox-4/pox_DelegateStxCommand.ts

```diff
@@ -0,0 +1,116 @@
+import { PoxCommand, Real, Stub, Wallet } from "./pox_CommandModel.ts";
+import { poxAddressToTuple } from "@stacks/stacking";
+import { expect } from "vitest";
+import { boolCV, Cl } from "@stacks/transactions";
+
+/**
+ * The `DelegateStxCommand` delegates STX for stacking within PoX-4. This operation 
+ * allows the `tx-sender` (the `wallet` in this case) to delegate stacking participation 
+ * to a `delegatee`.
+ *
+ * Constraints for running this command include:
+ * - The Stacker cannot currently be a delegator in another delegation.
+ * - The PoX address provided should have a valid version (between 0 and 6 inclusive).
+ */
+export class DelegateStxCommand implements PoxCommand {
+  readonly wallet: Wallet;
+  readonly delegateTo: Wallet;
+  readonly untilBurnHt: number;
+  readonly amount: bigint;
+
+  /**
+   * Constructs a `DelegateStxCommand` to delegate uSTX for stacking.
+   *
+   * @param wallet - Represents the Stacker's wallet.
+   * @param delegateTo - Represents the Delegatee's STX address.
+   * @param untilBurnHt - The burn block height until the delegation is valid.
+   * @param amount - The maximum amount the `Stacker` delegates the `Delegatee` to
+   *                 stack on his behalf
+   */
+  constructor(
+    wallet: Wallet,
+    delegateTo: Wallet,
+    untilBurnHt: number,
+    amount: bigint,
+  ) {
+    this.wallet = wallet;
+    this.delegateTo = delegateTo;
+    this.untilBurnHt = untilBurnHt;
+    this.amount = amount;
+  }
+
+  check(model: Readonly<Stub>): boolean {
+    // Constraints for running this command include:
+    // - The Stacker cannot currently be a delegator in another delegation.
+    return (
+      model.stackingMinimum > 0 &&
+      !model.wallets.get(this.wallet.stxAddress)?.hasDelegated
+    );
+  }
+
+  run(model: Stub, real: Real): void {
+    // The amount of uSTX delegated by the Stacker to the Delegatee.
+    // Even if there are no constraints about the delegated amount, 
+    // it will be checked in the future, when calling delegate-stack-stx.
+    const amountUstx = Number(this.amount);
+
+    // Act
+    const delegateStx = real.network.callPublicFn(
+      "ST000000000000000000002AMW42H.pox-4",
+      "delegate-stx",
+      [
+        // (amount-ustx uint)
+        Cl.uint(amountUstx),
+        // (delegate-to principal)
+        Cl.principal(this.delegateTo.stxAddress),
+        // (until-burn-ht (optional uint))
+        Cl.some(Cl.uint(this.untilBurnHt)),
+        // (pox-addr (optional { version: (buff 1), hashbytes: (buff 32) }))
+        Cl.some(poxAddressToTuple(this.delegateTo.btcAddress)),
```

Comment:
@setzeus here is the delegated `pox address`. We can change this if we store the `delegatedPoxAddress` inside the `wallet` object, too @moodmosaic .